main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -585,6 +585,69 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
             case xs => WeededAst.Expression.Tuple(xs, mkSL(sp1, sp2))
           }
 
+        case ParsedAst.Expression.ArrayLit(sp1, elms, sp2) =>
+          @@(elms.map(e => visit(e, unsafe))) map {
+            case es => WeededAst.Expression.ArrayLit(es, mkSL(sp1, sp2))
+          }
+
+        case ParsedAst.Expression.ArrayNew(sp1, elm, len, sp2) =>
+          @@(visit(elm, unsafe), visit(len, unsafe)) map {
+            case(es, ln) => WeededAst.Expression.ArrayNew(es, ln, mkSL(sp1, sp2))
+          }
+
+        case ParsedAst.Expression.ArrayLoad(exp1, exp2, sp2) =>
+          val sp1 = leftMostSourcePosition(exp1)
+          val loc = mkSL(sp1, sp2)
+
+          @@(visit(exp1, unsafe), visit(exp2, unsafe)) map {
+            case (ex1, ex2) => WeededAst.Expression.ArrayLoad(ex1, ex2, loc)
+          }
+
+        case ParsedAst.Expression.ArrayStore(exp1, exp2, exp3, sp2) =>
+          val sp1 = leftMostSourcePosition(exp1)
+          val loc = mkSL(sp1, sp2)
+
+          @@(visit(exp1, unsafe), visit(exp2, unsafe), visit(exp3, unsafe)) map {
+            case(ex1, ex2, ex3) => WeededAst.Expression.ArrayStore(ex1, ex2, ex3, loc)
+          }
+
+        case ParsedAst.Expression.ArrayLength(sp1, exp, sp2) =>
+          visit(exp, unsafe) map {
+            case es => WeededAst.Expression.ArrayLength(es, mkSL(sp1, sp2))
+          }
+
+        case ParsedAst.Expression.ArraySlice(exp1, exp2, exp3, sp2) =>
+          val sp1 = leftMostSourcePosition(exp1)
+          val loc = mkSL(sp1, sp2)
+
+          @@(visit(exp1, unsafe), visit(exp2, unsafe), visit(exp3, unsafe)) map {
+            case(ex1, ex2, ex3) => WeededAst.Expression.ArraySlice(ex1, ex2, ex3, loc)
+          }
+
+        case ParsedAst.Expression.VecLit(sp1, elms, sp2) =>
+          /*
+           * Rewrites empty vectors to Unit.
+           */
+          @@(elms.map(e => visit(e, unsafe))) map{
+            case Nil =>
+              val loc = mkSL(sp1, sp2)
+              WeededAst.Expression.Unit(loc)
+            case es => WeededAst.Expression.VecLit(es, mkSL(sp1, sp2))
+          }
+
+//        case ParsedAst.Expression.VecNew(sp1, elm, len, sp2) =>
```

Comment:
Here you will need to check that vectors are only constructed with positive 32 bit integers.